### ISSUE TITLE
Last exercise not marked "DONE"

### ISSUE THREAD

**Issue by Nahor** [rust-lang/rustlings#2024]
(v6.0.1) After finishing all the exercises and restarting `restlings`, if one goes to the list view, the last exercise is still marked "pending" and the progress bar is incomplete (93/94)

**Comment by mo8it** [comment]
This is intentional so people can repeat exercises. One can just enter `n` to mark the exercise as "done" again :)

**Comment by Nahor** [comment]
> One can just enter n to mark the exercise as "done" again

That's the issue. I did do `n`, and rustlings ran a last check on all the exercises and quit with the congratulation screen. Then I restarted rustlings and that last exercise is NOT mark "done".

> This is intentional so people can repeat exercises.

Even if an exercise is marked "done", one can still do it repeat it, e.g. by using the "continue at" option in the list view.
And for that matter, when doing that "continue at", an exercise does not become "pending", even though it's the same thing as restarting rustlings after it finished, in which case, rustlings is kind of doing a "continue at [last exercise]".

---

For info, the same happen with a different exercise by doing the following:
- after completing the last exercise, go to the list view instead of pressing `n`
- while on the list view, modify a previous exercise so that it does not succeed anymore
- quit the list view (last exercise still selected) and press `n`
=> this will run the check on all the exercises, fail on the modified one and select it
- fix the modified exercise and press `n` again
=> this will run the check again, succeed and quit rustlings
- restart rustlings (the modified exercise will be automatically selected)
- go to the list view

Result: the last exercise (#94) will be mark "done", but the one that was modified will be "pending".
While this is consistent with the issue I raised, it is inconsistent with the normal behavior. Just because an exercise is that last one to be finished of the whole series, it does not get marked "done" when pressing `n`.

---
My expectation would be for rustlings to indeed select the last exercise that was completed, but for that exercise to be flagged "done" in the list view anyway (since it was finished and the user tried to go to the next exercise, but since there were none rustlings exited instead).
